The version taken from GnuPG 1.4 assumes any cross-compiled package have
prefixed underscores, which is not the case; by using libtool's own macro,
we can avoid the whole issue.

--- a/configure.ac
+++ b/configure.ac
@@ -1022,7 +1022,7 @@
 # Check whether underscores in symbols are required.  This needs to be
 # done before setting up the assembler stuff.
 #
-GNUPG_SYS_SYMBOL_UNDERSCORE()
+LT_SYS_SYMBOL_USCORE
 
 
 #################################
--- a/mpi/config.links
+++ b/mpi/config.links
@@ -362,7 +362,7 @@
 
 # Make sysdep.h
 echo '/* created by config.links - do not edit */' >./mpi/sysdep.h
-if test x$ac_cv_sys_symbol_underscore = xyes; then
+if test x$sys_symbol_underscore = xyes; then
     cat <<EOF >>./mpi/sysdep.h
 #if __STDC__
 #define C_SYMBOL_NAME(name) _##name
